lib/Hooks.re

```diff
@@ -116,25 +110,10 @@ module State = {
 
     let onStateDidChange = hooks.onStateDidChange;
 
-    let setter = nextState =>
-      if (stateContainer.stale) {
-        let backtrace = Printexc.get_backtrace();
-        Printf.printf(
-          "
-          WARNING: A stale state setter has been used. The state has been updated and flushed since using state hook here:\n
-          \n
-          %s
-          \n
-          Using stale setters might lead to race conditions which are hard to debug. If you want to update state
-          asynchronously without updating the setter function each time, we recommend using the reducer hook
-          which enforces handling of race conditions.
-          ",
-          backtrace,
-        );
-      } else {
-        setState(nextState, stateContainer);
-        onStateDidChange();
-      };
+    let setter = updater => {
+      setState(updater(stateContainer.currentValue), stateContainer);
```

Comment:
I think the updater should always pass `nextValue`.

Reply:
Yeah, that makes sense. Done.